rtsp.go

```diff
@@ -913,41 +910,39 @@ func isCompactableH264(nalu []byte) bool {
 }
 
 // Image returns the latest frame as JPEG bytes.
-func (rc *rtspCamera) Image(_ context.Context, _ string, _ map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
-	if videoCodec(rc.currentCodec.Load()) == MJPEG {
-		return rc.getMJPEGImage()
+func (rc *rtspCamera) Image(_ context.Context, mimeType string, _ map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
+	// For now, we only support JPEG
+	if mimeType != rutils.MimeTypeJPEG && mimeType != rutils.WithLazyMIMEType(rutils.MimeTypeJPEG) {
+		rc.logger.Warn("unsupported mime type request %s, returning JPEG", mimeType)
 	}
-	return rc.getFrameAsImage()
-}
-
-// getMJPEGImage retrieves the latest MJPEG image.
-func (rc *rtspCamera) getMJPEGImage() ([]byte, camera.ImageMetadata, error) {
-	latestImg := rc.latestMJPEGImage.Load()
-	if latestImg == nil {
-		return nil, camera.ImageMetadata{}, errors.New("no mjpeg frame yet")
+	if videoCodec(rc.currentCodec.Load()) == MJPEG {
+		mjpegBytes := rc.latestMJPEGBytes.Load()
+		if mjpegBytes == nil {
+			return nil, camera.ImageMetadata{}, errors.New("no frame yet")
+		}
+		return *mjpegBytes, camera.ImageMetadata{
+			MimeType: rutils.MimeTypeJPEG,
+		}, nil
 	}
```

Comment:
This is where we'd add Yuv frame handling?

Reply:
Let's make it a case structure from the onset if so

Reply:
ok added switch case template.